src/Fantomas.Core.Tests/FormatAstTests.fs

```diff
@@ -137,3 +137,7 @@ let ``object expression should preserve override keyword on AST formatting with
         equal
         """{ new System.IDisposable with
     override __.Dispose() = () }"""
+
+[<Test>]
+let ``character quotes should be preserved, 3076`` () =
+    formatAstWithSourceCode "let x = 'A'" |> should equal "let x = 'A'"
```

Comment:
I was unsure where to add the test. Let me know what you think the best place would be.

Interestingly, this test was immediately passing, so `CodeFormatter.ParseAsync(false, source = sourceCode)` and then `CodeFormatter.FormatASTAsync(ast, source = sourceCode)` does not seem to have the same issue that using `parseFile` and then `CodeFormatter.FormatASTAsync` has.

Reply:
Would it make sense to remove this test then?

Reply:
Yeah, it doesn't quite your changes.